src/button/ReactionToggleButton.tsx

```
@@ -0,0 +1,324 @@
+        <section>
+          <li key="search" className={styles.reactionPopupMenuItem}>
+            <Tooltip label="Search">
+              <CpdButton
+                iconOnly
+                aria-label="Open reactions search"
+                Icon={SearchIcon}
+                kind="tertiary"
+                onClick={() => setIsSearching(true)}
+              />
+            </Tooltip>
+          </li>
+        </section>
+      ) : null}
+    </div>
+  );
+}
+
+interface ReactionToggleButtonProps {
+  rtcSession: MatrixRTCSession;
+  client: MatrixClient;
+}
+
+export function ReactionToggleButton({
+  client,
+  rtcSession,
+}: ReactionToggleButtonProps): ReactNode {
+  const { raisedHands, lowerHand, reactions } = useReactions();
+  const [busy, setBusy] = useState(false);
+  const userId = client.getUserId()!;
+  const isHandRaised = !!raisedHands[userId];
+  const memberships = useMatrixRTCSessionMemberships(rtcSession);
+  const [showReactionsMenu, setShowReactionsMenu] = useState(false);
+
+  const canReact = !reactions[userId];
+
+  const sendRelation = useCallback(
+    async (reaction: ReactionOption) => {
+      const myMembership = memberships.find((m) => m.sender === userId);
+      if (!myMembership?.eventId) {
+        logger.error("Cannot find own membership event");
+        return;
+      }
+      const parentEventId = myMembership.eventId;
+      try {
+        setBusy(true);
+        await client.sendEvent(
+          rtcSession.room.roomId,
+          ElementCallReactionEventType,
+          {
+            "m.relates_to": {
+              rel_type: RelationType.Reference,
+              event_id: parentEventId,
+            },
+            emoji: reaction.emoji,
+            name: reaction.name,
+          },
+        );
+        // Do NOT close the menu after this.
+      } catch (ex) {
+        logger.error("Failed to send reaction", ex);
+      } finally {
+        setBusy(false);
+        setShowReactionsMenu(false);
+      }
```

Looks like this `finally` block is always closing the menu ..?

---

Yeah, I don't know how best to present a failure right now. Asking in standup.

---

Have added a generic error box so that we can raise the error state.